lib/Client.php

```diff
@@ -285,7 +285,7 @@ private static function onEmptyRead($state) {
             $finalResult = isset($op->buffer[0]) ? $op->buffer : null;
             $op->promisor->succeed($finalResult);
             foreach ($state->readOperations as $op) {
-                $op->succeed();
+                $op->promisor->succeed();
             }
```

Comment:
Not sure I follow. The `succeed()` before the foreach is on an operation that was shifted from the `readOperations` array.